libs/wire-api/src/Wire/API/Routes/Version.hs

```diff
@@ -57,44 +57,63 @@ import Servant
 import Servant.Swagger
 import Wire.API.Routes.Named
 import Wire.API.VersionInfo
+import Wire.Arbitrary (Arbitrary, GenericUniform (GenericUniform))
 
 -- | Version of the public API.
 data Version = V0 | V1 | V2 | V3
-  deriving stock (Eq, Ord, Bounded, Enum, Show)
+  deriving stock (Eq, Ord, Bounded, Enum, Show, Generic)
   deriving (FromJSON, ToJSON) via (Schema Version)
+  deriving (Arbitrary) via (GenericUniform Version)
+
+versionString :: IsString a => Version -> a
+versionString V0 = "v0"
```

Comment:
I think this could be implemented with [`fromEnum`](https://hoogle.zinfra.io/file/nix/store/wnv3042jbcai41bm9djjbkgfxlwx0nay-ghc-9.2.4-doc/share/doc/ghc/html/libraries/ghc-9.2.4/GHC-Prelude.html#v:fromEnum).

The benefit would be that you won't have to touch this function when a new version is introduced. The drawback is that `Version` constructors have to be sorted.

Reply:
i used to agree, but been there!  :)

- https://github.com/wireapp/wire-server/pull/2978#pullrequestreview-1247230615
- d2e52a1bf0f94c5d5dbbc80d1443100a54e7c1af

Reply:
This simple version is fine with me, too :smile_cat: